job/job.go

```diff
@@ -50,6 +91,20 @@ func CreateFromDHCP(mac net.HardwareAddr, giaddr net.IP, circuitID string) (Job,
 	if err != nil {
 		return Job{}, err
 	}
+
+	// check if there is an active workflow for the device
+	hd := (*d).Hardware()
+	hwID := (*hd).HardwareID()
+	activeWorkflows, err := isHardwareHavingActiveWorkflow(hwID)
+	if err != nil {
+		return Job{}, err
+	}
+
+	if !activeWorkflows {
+		s := fmt.Sprintf("no active workflow found for hardware %s", hwID)
+		return Job{}, errors.New(s)
```

Comment:
can be combined into just `errors.Errorf`

Reply:
done